pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/LLCRealtimeClusterIntegrationTest.java

```diff
@@ -271,7 +271,17 @@ public void testAddRemoveDictionaryAndInvertedIndex()
     JsonNode queryResponse = postQuery(query);
     assertEquals(queryResponse.get("totalDocs").asLong(), numTotalDocs);
     // Full table scan without dictionary
-    assertEquals(queryResponse.get("numEntriesScannedInFilter").asLong(), numTotalDocs);
+    // The offline segments are created deterministically and all the offline segments contain "ActualElapsedTime =
+    // -9999" records.
+    // The realtime segments are created non-deterministically (uses system time for Kafka partitioning) and there is
```

Comment:
Would it be a better fix to make the data partitioning more deterministic for realtime tests?

Reply:
Done.